src/Umbraco.Core/Cache/Refreshers/Implement/ContentCacheRefresher.cs

```diff
@@ -172,9 +172,23 @@ private void HandleMemoryCache(JsonPayload payload)
                 var branchKeys = descendantsKeys.ToList();
                 branchKeys.Add(key);
 
-                foreach (Guid branchKey in branchKeys)
+                // If unpublished cultures has one or more values, but published cultures does not, this means that the branch is unpublished entirely
+                // And therefore should no longer be resolve-able from the cache, so we need to remove it instead.
+                // Otherwise, some culture is still published, so it should be resolve-able from cache, and published cultures should instead be used.
+                if (payload.UnpublishedCultures is not null && payload.UnpublishedCultures.Length != 0 &&
```

Comment:
Maybe move this into a small helper method, just so you can name it something like `IsBranchUnpublished` or similar meaningful name, and would remove this CodeScene warning.  Please keep the comment though, it's useful.

Reply:
That's an excellent point, thanks, much cleaner :tada: